src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -232,43 +481,110 @@ public Authentication authenticate(Authentication authentication) throws Authent
+                        for (String userGroup : usersGroups) {                    	
+                        	// copy any of the user's permissions from the prior auth mgr to our new one
+                        	for (PermissionGroup pg : permissionGroups) {
+                        		for (Permission p : pg.getPermissions()) {
+                        			if (existingAuthMgr.hasPermission(userGroup, p)) {
+                        				newAuthMgr.add(p, userGroup);
+                        			}
+                        		}
+                        	}
+                        }
+                        
+                    	if (addInsteadOfCopy) {
+                        	if (LOGGER.isLoggable(Level.FINE))
+                        		LOGGER.fine(String.format("onSuccess: adding permissions to new user %s based on openshift roles %s", info.getName(), allowedRoles));
+                        	
+                        	// map OpenShift user based on role to Jenkins user with analogous permissions
+                        	//TODO can we minimally assume view role ?  assuming so while waiting for the build.oenshift.io.jenkins related changes we think are coming
+                        	// and see if a self-sar against the view role comes back and says allowed (currently returns allowed as false for the view role)
+                        	newAuthMgr.add(Hudson.READ, info.getName());
+                        	newAuthMgr.add(Item.READ, info.getName());
+                			if (allowedRoles.contains("edit") || allowedRoles.contains("admin")) {
+                				newAuthMgr.add(Item.BUILD, info.getName());
+                				newAuthMgr.add(Item.CONFIGURE, info.getName());
+                				newAuthMgr.add(Item.CREATE, info.getName());
+                				newAuthMgr.add(Item.DELETE, info.getName());
+                				newAuthMgr.add(Item.WORKSPACE, info.getName());
+                				newAuthMgr.add(SCM.TAG, info.getName());
+                			}
+                			if (allowedRoles.contains("admin")) {
+                				newAuthMgr.add(Computer.CONFIGURE, info.getName());
+                				newAuthMgr.add(Computer.DELETE, info.getName());
+                				newAuthMgr.add(Hudson.ADMINISTER, info.getName());
+                				newAuthMgr.add(Hudson.READ, info.getName());
+                				newAuthMgr.add(Run.DELETE, info.getName());
+                				newAuthMgr.add(Run.UPDATE, info.getName());
+                				newAuthMgr.add(View.CONFIGURE, info.getName());
+                				newAuthMgr.add(View.CREATE, info.getName());
+                				newAuthMgr.add(View.DELETE, info.getName());
+                			}
+                    	}
+            			
+            			Jenkins.getInstance().setAuthorizationStrategy(newAuthMgr);
```

Comment:
I'm a little concerned about refreshing tied to the session.  I don't like
polling, and I'm not sure that we need rapid reaction to permissions in
openshift.  What other options do we have?

On Fri, Aug 26, 2016 at 4:52 PM, Gabe Montero <notifications@github.com>
wrote:

> In src/main/java/org/openshift/jenkins/plugins/openshiftlogin/
> OpenShiftOAuth2SecurityRealm.java
> <https://github.com/openshift/jenkins-openshift-login-plugin/pull/2#discussion_r76486201>
> :
>
> > +                				newAuthMgr.add(SCM.TAG, info.getName());
> > +                			}
> > +                			if (allowedRoles.contains("admin")) {
> > +                				newAuthMgr.add(Computer.CONFIGURE, info.getName());
> > +                				newAuthMgr.add(Computer.DELETE, info.getName());
> > +                				newAuthMgr.add(Hudson.ADMINISTER, info.getName());
> > +                				newAuthMgr.add(Hudson.READ, info.getName());
> > +                				newAuthMgr.add(Run.DELETE, info.getName());
> > +                				newAuthMgr.add(Run.UPDATE, info.getName());
> > +                				newAuthMgr.add(View.CONFIGURE, info.getName());
> > +                				newAuthMgr.add(View.CREATE, info.getName());
> > +                				newAuthMgr.add(View.DELETE, info.getName());
> > +                			}
> > +                    	}
> > +            			
> > +            			Jenkins.getInstance().setAuthorizationStrategy(newAuthMgr);
>
> OK, I've pushed the change with the servlet filter so that we poll
> openshift for updated permissions as the given user traverses through
> jenkins.
>
> I've initially declined to expose the polling interval as a config param.
> My hope is there is an agreeable default and the user has 1 less tuning
> know to digest. But of course chime in as desired.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/jenkins-openshift-login-plugin/pull/2/files/d32f29cf646040cbf44ea259d66012cdf6011e6c#r76486201>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_pygj62nA-HymGZpRXmrCMA5HBkEfks5qj1H4gaJpZM4Jfdh->
> .
>